traffic_portal/app/src/modules/public/login/LoginController.js

```diff
@@ -48,9 +53,31 @@ var LoginController = function($scope, $log, $uibModal, authService, userService
         });
     };
 
+    $scope.loginOauth = function() {
+        var continueURL = '';
+        const gotoUrl = propertiesModel.properties.oAuth.oAuthUrl;
+        const gotoUrlWithoutParams = gotoUrl.split('?')[0];
```

Comment:
though it's odd, a `?` could conceivably appear in the path of a URL and not necessarily signify the beginning of query params - if you use a `URL` object you can get what you're looking for with `urlObject.origin + urlObject.pathname` (optionally support document fragments by appending `# + urlObject.hash` if desired/if it exists).

Reply:
done